src/rust/cryptography-x509-verification/src/policy/extension.rs

```diff
@@ -2,121 +2,62 @@
 // 2.0, and the BSD License. See the LICENSE file in the root of this repository
 // for complete details.
 
-use cryptography_x509::oid::{
-    AUTHORITY_INFORMATION_ACCESS_OID, AUTHORITY_KEY_IDENTIFIER_OID, BASIC_CONSTRAINTS_OID,
-    EXTENDED_KEY_USAGE_OID, KEY_USAGE_OID, NAME_CONSTRAINTS_OID, SUBJECT_ALTERNATIVE_NAME_OID,
-    SUBJECT_KEY_IDENTIFIER_OID,
-};
 use cryptography_x509::{
     certificate::Certificate,
     extensions::{Extension, Extensions},
 };
+use std::collections::{HashMap, HashSet};
 
 use crate::{ops::CryptoOps, policy::Policy, ValidationError};
 
 pub(crate) struct ExtensionPolicy<B: CryptoOps> {
-    pub(crate) authority_information_access: ExtensionValidator<B>,
-    pub(crate) authority_key_identifier: ExtensionValidator<B>,
-    pub(crate) subject_key_identifier: ExtensionValidator<B>,
-    pub(crate) key_usage: ExtensionValidator<B>,
-    pub(crate) subject_alternative_name: ExtensionValidator<B>,
-    pub(crate) basic_constraints: ExtensionValidator<B>,
-    pub(crate) name_constraints: ExtensionValidator<B>,
-    pub(crate) extended_key_usage: ExtensionValidator<B>,
+    pub(crate) validator_by_oid: HashMap<asn1::ObjectIdentifier, ExtensionValidator<B>>,
 }
 
 impl<B: CryptoOps> ExtensionPolicy<B> {
+    pub(crate) fn new(
+        validator_by_oid: HashMap<asn1::ObjectIdentifier, ExtensionValidator<B>>,
+    ) -> Self {
+        Self { validator_by_oid }
```

Comment:
The motivation is to be able to support custom extension types. I was initially planning to limit the scope of user-customizable extension verification to user provided callbacks, but @woodruffw  said he would prefer to allow custom extension OIDs since they have a usecase for that (see this comment https://github.com/pyca/cryptography/issues/11165#issuecomment-2326724920)

If performance is really such an issue with this implementation and we still want to keep the ability to define validators for arbitrary extensions, we could potentially just leave the default extension validation as is and add a map for custom extension validators.

Reply:
Yes, I think named validators for the standard ones + a map for extending it sounds good.

Reply:
Alright. What I'm not sure about though is if I should once again wait until the meat of the feature is implemented so the coverage for this is free, or if I should add some tests specific to this to the rust crate and make the PR now.

Reply:
I think either approach could work, so whichever you prefer is fine. But if it were me, I'd implement the whole feature, and then if the overall patch is too large we can look at ways to split pieces out.